packages/cli/test/integration/users.endpoints.test.ts

```diff
@@ -0,0 +1,491 @@
+		firstName: utils.randomName(),
+		lastName: utils.randomName(),
+		password: utils.randomValidPassword(),
+	});
+
+	expect(response.statusCode).toBe(400);
+});
+
+test('POST /users should fail if emailing is not set up', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	const response = await authOwnerAgent.post('/users').send([{ email: utils.randomEmail() }]);
+
+	expect(response.statusCode).toBe(500);
+});
+
+test('POST /users should email invites and create user shells', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	const {
+		user,
+		pass,
+		smtp: { host, port, secure },
+	} = await utils.getSmtpTestAccount();
+
+	config.set('userManagement.emails.mode', 'smtp');
+	config.set('userManagement.emails.smtp.host', host);
+	config.set('userManagement.emails.smtp.port', port);
+	config.set('userManagement.emails.smtp.secure', secure);
+	config.set('userManagement.emails.smtp.auth.user', user);
+	config.set('userManagement.emails.smtp.auth.pass', pass);
+
+	const payload = TEST_EMAILS_TO_CREATE_USER_SHELLS.map((e) => ({ email: e }));
+
+	const response = await authOwnerAgent.post('/users').send(payload);
+
+	expect(response.statusCode).toBe(200);
+
+	for (const { id, email: receivedEmail } of response.body.data) {
+		expect(validator.isUUID(id)).toBe(true);
+		expect(TEST_EMAILS_TO_CREATE_USER_SHELLS.some((e) => e === receivedEmail)).toBe(true);
+
+		const user = await Db.collections.User!.findOneOrFail(id);
+		const { firstName, lastName, personalizationAnswers, password, resetPasswordToken } = user;
+
+		expect(firstName).toBeNull();
+		expect(lastName).toBeNull();
+		expect(personalizationAnswers).toBeNull();
+		expect(password).toBeNull();
+		expect(resetPasswordToken).toBeNull();
+	}
+});
+
+test('POST /users should fail with invalid inputs', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	config.set('userManagement.emails.mode', 'smtp');
+
+	const invalidPayloads = [
+		utils.randomEmail(),
+		[utils.randomEmail()],
+		{},
+		[{ name: utils.randomName() }],
+		[{ email: utils.randomName() }],
+	];
+
+	for (const invalidPayload of invalidPayloads) {
+		const response = await authOwnerAgent.post('/users').send(invalidPayload);
+		expect(response.statusCode).toBe(400);
+	}
+});
+
+test('POST /users should ignore an empty payload', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	config.set('userManagement.emails.mode', 'smtp');
+
+	const response = await authOwnerAgent.post('/users').send([]);
+
+	const { data } = response.body;
+
+	expect(response.statusCode).toBe(200);
+	expect(Array.isArray(data)).toBe(true);
+	expect(data.length).toBe(0);
```

Comment:
we should also check if the DB is still empty

Reply:
Not empty but having 1 user, the owner.